### PR TITLE
323  [Bug]: Uncaught error in Barplot module

### LINKED ISSUE #323: [Bug]: Uncaught error in Barplot module
### What happened?

In [RNA-seq app](https://genentech.shinyapps.io/nest_rna-seq_main/):

- Barplot
- Select Experiment - hd2
- Selected Assay - any
- Optional facet variable - any
- Selected Genes - any
- Select Gene Summary - any
- Select Quantiles - any
- Filter Panel - select any filter that reduces size of the data

`Error: 5 elements in value to replace 4 elements`
(numbers may vary)

It stems from `multiSampleVarSpecServer(...)$experiment_data()` call _(see line below)_:

https://github.com/insightsengineering/teal.modules.hermes/blob/11ee3086b0be214d264e98d80d79eb91c1f80383/R/barplot.R#L151

If this error is expected, use a validation mechanism to capture it.

This error is probably related to #320 

#### Screenshot

image

### PR DESCRIPTION
**Pull request by BFalquet**
close #323 

Add missing reactivity element that prevents correct update when data is filtered.

thank you for the review

### CODE REVIEW

**Review by danielinteractive — APPROVED** [review]
Thanks @BFalquet ! Maybe add a news entry

**Review by gogonzo — CHANGES_REQUESTED** [review]
WIP comments

File: R/sampleVarSpec.R [hunk 1]
```diff
@@ -284,7 +284,11 @@ sampleVarSpecServer <- function(id, # nolint
   assert_string(label_modal_title)
 
   moduleServer(id, function(input, output, session) {
-    start_col_data <- eventReactive(experiment_name(), {
+
+    to_observe <- reactive({list(experiment_name(), original_data())})
+
+    start_col_data <- eventReactive(to_observe(), {
```
Comment (gogonzo) [inline R/sampleVarSpec.R:290]: Doesn't have to trigger on `experiment_name`. `experiment_name` isn't used in this reactive and also not needed to keep module working
```suggestion
    start_col_data <- reactive({
```

Comutation of `final_data` depends on several reactives and `current_combination` which depends on `experment_name`. Red link denotes that not needed

merge_data-Page-4

Comment (BFalquet) [reply R/sampleVarSpec.R:290]: You are right, I am now a bit confused about the original design.

Comment (danielinteractive) [reply R/sampleVarSpec.R:290]: I am not so sure. The col data variables generally depend on which experiment was selected. Maybe not in the current data example in the app but in general they can be different

**Review by danielinteractive — APPROVED** [review]

**Pull request merged**